bool ball_detc
bool ball_reach
